pandas/io/formats/csvs.py

```diff
@@ -172,10 +172,13 @@ def save(self) -> None:
             not hasattr(self.path_or_buf, "write") and self.compression == "zip"
         )
 
-        if is_zip:
-            # zipfile doesn't support writing string to archive. uses string
-            # buffer to receive csv writing and dump into zip compression
-            # file handle. GH21241, GH21118
+        uses_buffer = is_zip or (
+            hasattr(self.path_or_buf, "write") and "b" in self.mode
```

Comment:
can use is_binary here?

Reply:
I defined `is_binary` in a different function but I split the statement to make it maybe a bit more clear